### PR TITLE
Clean up redundant bits in CI

### PR DESCRIPTION
**Pull request by crusaderky**

### CODE REVIEW

File: continuous_integration/environment-3.12.yaml [hunk 1]
```diff
@@ -35,10 +35,10 @@ dependencies:
   # https://github.com/dask/dask/pull/9569
   # - tiledb-py # crashes on Python 3.11
   # - pyspark
-  # - tiledb>=2.5.0 # crashes on Python 3.11
+  # - tiledb # crashes on Python 3.11
   - xarray
-  - sqlalchemy>=1.4.16
-  - pyarrow>=13
+  - sqlalchemy
+  - pyarrow
```
Comment (phofl) [inline continuous_integration/environment-3.12.yaml:41]: Please keep the arrow versions restricted to ensure that we test things like the convert string options

Comment (crusaderky) [reply continuous_integration/environment-3.12.yaml]: OK, I've changed the 3.11 recipe from `pyarrow>=11` to `pyarrow>=13`. The former IMHO caused confusion as the recipe installs 13 anyway.

**Review by phofl — APPROVED** [review]

File: continuous_integration/environment-3.12.yaml (hunk 1 above)
Comment (phofl) [reply continuous_integration/environment-3.12.yaml]: Fine with me as long as we have a lower pin that forces the string option

**Comment by phofl** [comment]
thx @crusaderky

**Pull request merged**